### PR TITLE
Enable auth config on Milvus resource

### LINKED ISSUE #4966: Modify Milvus hosting resource to use generated password by default
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Miluvs changed to support config-based password default (instead of global default) (see: https://github.com/milvus-io/milvus/issues/33058). This bug tracks modifying reacting to that change in the Aspire.Hosting.Milvus package to enable a generated default password if none provided, similar to other resources.  This will only work on a later tagged image of the milvus container image. /cc @eerhardt

### PR DESCRIPTION
**Pull request by timheuer**
- Updates Milvus hosting to leverage dynamically generated default password if none provided (similar to other resources) and resulting cascading changes + tests.
- Updates tags for contianer image that support setting auth

## Description

Milvus added the ability to set the root password. This change adapts to that and follows similar pattern to other config-based auth (e.g., MySql is a good analog). This requires the support of a minimum new Milvus version (2.4.7). 

Fixes #4966

## Checklist

- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [x] No. Follow-up changes expected. - haven't seen a pattern where we've made a change that requires a specific new resource version
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [x] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [x] No - follows same pattern as existing pattern/resources
  - [ ] No
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Link to aspire-docs issue: [dotnet/docs-aspire#1475](https://github.com/dotnet/docs-aspire/issues/1475)
  - [ ] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5229)

### CODE REVIEW

File: src/Aspire.Hosting.Milvus/MilvusBuilderExtensions.cs [hunk 1]
```diff
@@ -41,13 +41,13 @@ public static class MilvusBuilderExtensions
     /// <returns>A reference to the <see cref="IResourceBuilder{MilvusServerResource}"/>.</returns>
     public static IResourceBuilder<MilvusServerResource> AddMilvus(this IDistributedApplicationBuilder builder,
         string name,
-        IResourceBuilder<ParameterResource> apiKey,
+        IResourceBuilder<ParameterResource>? apiKey = null,
         int? grpcPort = null)
     {
-        ArgumentNullException.ThrowIfNull(apiKey, nameof(apiKey));
+        var apiKeyParameter = apiKey?.Resource ??
+            ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-Key", special: false);
```
Comment (eerhardt) [inline src/Aspire.Hosting.Milvus/MilvusBuilderExtensions.cs:48]: Why `special: false`?

Comment (eerhardt) [reply src/Aspire.Hosting.Milvus/MilvusBuilderExtensions.cs:48]: (super nit): in other places we use lower case names: `$"{name}-password"` but here we have PascalCase `-Key`. Should we be consistent?

File: tests/Aspire.Hosting.Tests/Milvus/AddMilvusTests.cs [hunk 2]
```diff
@@ -89,7 +89,7 @@ public async Task MilvusCreatesConnectionString()
         var connectionStringResource = milvus.Resource as IResourceWithConnectionString;
 
         var connectionString = await connectionStringResource.GetConnectionStringAsync();
-        Assert.Equal($"Endpoint=http://localhost:19530;Key=pass", connectionString);
+        Assert.Equal($"Endpoint=http://localhost:19530;Key=root:pass", connectionString);
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Tests/Milvus/AddMilvusTests.cs:92]: These tests have all moved out of `Aspire.Hosting.Tests` and are in `Aspire.Hosting.Milvus.Tests` now. Can you sync main?

Comment (timheuer) [reply tests/Aspire.Hosting.Tests/Milvus/AddMilvusTests.cs:92]: doh, lots of changes, forgot to rebase...I will do that and resubmit I think. -- keep feedback coming and i'll address.

**Pull request closed without merging**